### PR TITLE
Modified save argument of plot_model()

### PR DESCRIPTION
**Pull request by bhanuteja2001**
## Enchancement #1288 
  - Enhanced the plot_model function to save the figure at any destination in the system. 

#### Describe the changes you've made
Modified the save argument of plot_model to save the figure at the destination mentioned by the user. Instead of just saving the figure in the current directory.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

Tested with the confusion matrix plot.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots

 Original picture of Log file         | Picture saved at the destination mentioned
 :--------------------: |:--------------------:
 **Screenshot (422)_LI**  | <b>
Screenshot (418)
 </b> |
 
Screenshot (420)

### CODE REVIEW

**Comment by Yard1** [comment]
Hey @bhanuteja2001 thank you for doing this. Could we do something like this instead?:
```python
        if isinstance(save, str):
            visualizer.show(outpath=f"{name}.png", clear_figure=True)
        else:
            logger.info(f"Saving '{name}.png' in the path specified")
            plt.savefig("{}/{}.png".format(save,name))

**Comment by bhanuteja2001** [comment]
@Yard1  sure let's do it this way

**Comment by bhanuteja2001** [comment]
@Yard1 I have made the changes.
Please check now.

File: pycaret/internal/tabular.py [hunk 1]
```diff
@@ -6918,8 +6934,10 @@ def tree():
             display.move_progress()
             display.clear_output()
             if save:
-                logger.info(f"Saving '{plot_name}.png' in current active directory")
-                plt.savefig(f"{plot_name}.png", bbox_inches="tight")
+                    if not isinstance(save, bool):
```
Comment (Yard1) [inline pycaret/internal/tabular.py:6937]: Indentation

File: pycaret/internal/tabular.py [hunk 2]
```diff
@@ -6963,8 +6981,10 @@ def calibration():
...
```
Comment (Yard1) [inline pycaret/internal/tabular.py:6984]: Indentation

File: pycaret/internal/tabular.py [hunk 3]
```diff
@@ -7607,7 +7631,10 @@ def summary(show: bool = True):
             shap_plot = shap.summary_plot(shap_values, test_X, show=show, **kwargs)
 
         if save:
-            plt.savefig(f"SHAP {plot}.png", bbox_inches="tight")
+            if save == True:
```
Comment (Yard1) [inline pycaret/internal/tabular.py:7634]: Missed here

File: pycaret/internal/tabular.py [hunk 4]
```diff
@@ -7640,7 +7667,10 @@ def correlation(show: bool = True):
             logger.info("model type detected: type 2")
             shap.dependence_plot(dependence, shap_values, test_X, show=show, **kwargs)
         if save:
-            plt.savefig(f"SHAP {plot}.png", bbox_inches="tight")
+            if save == True:
```
Comment (Yard1) [inline pycaret/internal/tabular.py:7670]: Missed here

**Review by Yard1 — CHANGES_REQUESTED** [review]
Thanks! Just a few more comments. Happy to merge once those are addressed and CI passes 🚀

**Comment by bhanuteja2001** [comment]
@Yard1 please check now.
I have made the changes as requested.

**Review by Yard1 — APPROVED** [review]

**Pull request merged**